### PR TITLE
Merge duplicate instructions for setting up build environment.

### PR DESCRIPTION
**Pull request by mkcor**
## Description

I'm following up on my https://github.com/scikit-image/scikit-image/issues/6748#issuecomment-1429280343. With this PR, there is now a single place for the build env setup instructions (so we don't need to maintain two versions in parallel). I've kept the version that worked for me, anticipating in particular the application period for Outreachy projects.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

**Review by stefanv — APPROVED** [review]
A big improvement, thanks!

I'm not sure how useful it is to link to `meson.md`, but it's probably OK for now.

File: INSTALL.rst [hunk 1]
```diff
@@ -327,12 +334,20 @@ before you get started.
...
+  ./dev.py build
   # Test your installation
-  pytest --pyargs skimage
+  ./dev.py test
+  # Build docs
+  ./dev.py docs
+  # Try the new version in IPython
+  ./dev.py ipython
+
+For more information about building and using the dev.py package, see ``meson.md``.
```
Comment (mkcor) [inline INSTALL.rst:350]: Thanks, @stefanv! Did you mean replacing ``meson.md`` with a link to the file?

**Review by jarrodmillman — APPROVED** [review]

File: INSTALL.rst (hunk 1 above)
Comment (stefanv) [reply INSTALL.rst:350]: Oh, I just meant that those were working notes, and not truly intended for mass consumption.

Comment (mkcor) [reply INSTALL.rst:350]: Oh, okay! I'm happy to remove it if it's preferable... Note that I didn't add this line: I just moved it from `CONTRIBUTING.rst` line 90 to `INSTALL.rst` line 350.

**Comment by stefanv** [comment]
Thanks, Marianne!

**Pull request merged**